_posts/2020-08-26-bazel-3.5.md

```diff
@@ -0,0 +1,108 @@
+---
+layout: posts
+title: "Bazel 3.5"
+authors:
+  - aiuto
+---
+
+[Bazel 3.5](https://github.com/bazelbuild/bazel/releases/tag/3.5.0) has just been released.
+
+Bazel 3.5 is a minor release. It is an incremental update to Bazel 3.4.
+
+Basline: 889bc0b523b47eeb38a72bf9bb6858ee525a7c7e
+
+Cherry picks:
+-   a7a0d48fbeb059ee60e77580e5d05baeefdd5699: Make no-op starlark transition not affect the output directory.
+-   b37c51c7085f0aefe04034dd451acb847605ddb5: Add include_prefix and strip_include_prefix to cc_common.compile
+-   f6ad35fcde93f92c591778ed7db38d167f5bbc03: Delete --experimental_transparent_compression
+-   39bc97eab295bddb35b38bfc4a2ff3d2b15d034e: Remove --experimental_action_args
+-   b9706675a7abf6ceebb250f0b3dfa4087a0c35f6: Stop needlessly parsing WORKSPACE files from external repositories.
+-   e574d558da17cfd0f818e7a937a07926aa270069: Allow hyphen char in workspace name
+-   9993785fa0c4fa4172aa31d306f3abea76833abf: Allow dot ('.') in workspace names.
+
+## Incompatible changes
+
+- Removed the `--experimental_process_wrapper_wait_fix` flag.
+- Removed the `--experimental_ui_deduplicate` flag.
+- Removed the `--experimental_transparent_compression` flag.
+- Removed the `--experimental_action_args` flag.
+- Bazel now correctly prefers Xcode versions in `/Applications` over
+  any other paths, which resolves an issue with accidentally picking up
+  an Xcode version from a Time Machine backup or network disk. In the
+  improbable case that you relied on the old behavior and Bazel now picks
+  up Xcode from the wrong location, you can fix it by moving that Xcode
+  version to /Applications.
+
+## New features:
+
+- `cquery` now follows aspects when the `--include_aspects` flag is used.
+- Add support to bazel/crosstool for building arm64 on macos.
+
+## Important changes:
+
+- Fixed resource shrinking when \<overlayable/> tags are used.
+- Remove legacy handling of `--extra_checks`
+- Remove old incompatible flag `--incompatible_symlinked_sandbox_expands_tree_artifacts_in_runfiles_tree`.
+- Update coverage configuration for Python, filegroup, and shell script
+  rules to distinguish between source and dependency attributes.
+
+- Add `InstrumentedFilesInfo` provider to Starlark globals.
+- Make filegroup always forward `InstrumentedFilesProvider` and not
+  collect any sources directly.
+- Major changes to reporting of Starlark errors and the call stack.
+  (Please be alert to possible regressions, such as errors that lack
+   relevant location information.)
+- Dot ('.') is now allowed in workspace names. See [#11837](https://github.com/bazelbuild/bazel/issues/11837).
+
+## Rules
```

Comment:
Should the Rules section be broken down by languages?

Reply:
It's not clear. We did not for 3.4, so I did not this time.  It seems each person does it their own way. Which indicates that we should either make this mechanical, or have the same editorial voice looking at all release notes and the release manager's job should only be to get it close, and then product management can make it right.

Reply:
We did for almost every release so far, because that's part of the process.
In 3.4, Tobias argued that it didn't make sense that time (there would be lots of sections with just one item). As far as I remember, that was resolved in a doc comment. So it's not "each person does it their own way".

Reply:
I did it anyway, but we're in the same situation as 3.4. Most sections have just one item.